src/utils/functionRuntimeUtils.ts

```diff
@@ -127,17 +120,22 @@ export namespace functionRuntimeUtils {
         return null;
     }
 
-    async function getNewestFunctionRuntimeVersion(major: number): Promise<string | null> {
-        const npmRegistryUri: string = 'https://aka.ms/W2mvv3';
-        type distTags = { core: string, docker: string, latest: string };
-        const distTags: distTags = <distTags>JSON.parse((await <Thenable<string>>request(npmRegistryUri).promise()));
-        switch (major) {
-            case FunctionRuntimeTag.latest:
-                return distTags.latest;
-            case FunctionRuntimeTag.core:
-                return distTags.core;
-            default:
-                return null;
+    async function getNewestFunctionRuntimeVersion(major: number, actionContext: IActionContext): Promise<string | undefined> {
+        try {
+            const npmRegistryUri: string = 'https://aka.ms/W2mvv3';
+            type distTags = { core: string, docker: string, latest: string };
+            const distTags: distTags = <distTags>JSON.parse((await <Thenable<string>>request(npmRegistryUri).promise()));
+            switch (major) {
+                case FunctionRuntimeTag.latest:
+                    return distTags.latest;
+                case FunctionRuntimeTag.core:
+                    return distTags.core;
+                default:
+            }
+        } catch (error) {
```

Comment:
Not really liking the idea of putting error messages into random telemetry properties. I'd rather we keep them standardized so that the current dashboard picks them up.  That suggests two possibilities (I'm looking at a similar scenario in ARM tools):

1) run this under a new telemetry event
2) run this under a new telemetry event, but only log the event on errors (via a new flag)

Thoughts?

Reply:
I generally agree with you, but I also don't like creating new events every time because it makes it harder to correlate properties together. I generally do it this way when I don't really care if the code fails (In this case, errors are never displayed to the user and worst case they don't get the prompt to update to the latest runtime).

I do this very infrequently, but this isn't the first time we do this in functions. I guess I don't have a super strong preference.

Reply:
Please comment on https://github.com/Microsoft/vscode-azuretools/issues/183.  For now we'll let this go.